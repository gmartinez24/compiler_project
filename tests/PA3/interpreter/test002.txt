// Tests not, relative ops, and call inside relation

main

{
    // Check not handling
    if(not(not(1==0))) then
        call printInt(-3);
    else
        call printInt(3);
    fi;

    // Check rel operators
    if(-1 < 1) then 
        call printInt(4);
    else
        call printInt(-4);
    fi;
    
    if(1 >= call readInt()) then
        call printInt(5);
    else
        call printInt(-5);
    fi;
}.