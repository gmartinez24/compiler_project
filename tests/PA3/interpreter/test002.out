3 4 int? 5 